### DISCUSSION TITLE
Is there a plugin for farm runs?

### DISCUSSION THREAD

**Discussion by ChopinDavid** [runelite/runelite#16713] [Ideas]
Hi,
I was wondering if there was a plugin for doing farm runs. That is, the plugin could calculate all of the patches you have available, what the best tree for each available patch is (given your farming level), what payments you need noted for your run, whether a tree was planted but not paid for, etc.

Does something like this exist? If not, I'll probably go ahead and create one.

I found [this](https://github.com/runelite/runelite/discussions/16533), but it seems to be a bit of a different idea. I'm not sure how good I'd be at calculating the best route for a run (in my head, I generally know how to get places), but I could see this being a future feature of the plugin after I get a 1.0.0 out.

**Comment by ChopinDavid** [comment]
cc: @kevinNLPG

**Comment by Speaax** [comment]
Closest thing I know is this: https://github.com/Speaax/Farming-Helper 
Called "Lazy Farming" on pluginhub. (self-promo xD)
Poorly written code, but personally I like the functionality and direction I'm going with it. 

Some features are missing or is lacking because I'm relatively new to coding, and have used GPT-4 to overcome a lot of my struggles. 

Yesterday I created a new project to redo a lot of the code so I can have better structure. You can join me if you want to, or take any part of my code, ask questions or whatever about this project.
